src/Repository/ConnectedGithubClient.php

```diff
@@ -0,0 +1,195 @@
+<?php
+
+namespace OAT\DependencyResolver\Repository;
+
+use Github\Exception\ErrorException;
+use Github\Exception\RuntimeException;
+use OAT\DependencyResolver\Repository\Entity\Repository;
+use OAT\DependencyResolver\Repository\Exception\BranchNotFoundException;
+use OAT\DependencyResolver\Repository\Exception\EmptyRepositoryException;
+use OAT\DependencyResolver\Repository\Exception\FileNotFoundException;
+
+class ConnectedGithubClient
```

Comment:
What does connected mean in this context? Do we have a non-connected client as well? I think `GithubClient` would be a better name